zebra-state/src/service/finalized_state/tests/vectors.rs

```diff
@@ -0,0 +1,40 @@
+use std::sync::Arc;
+
+use zebra_chain::{
+    block::{Block, Height},
+    parameters::Network,
+    serialization::ZcashDeserializeInto,
+};
+use zebra_test::prelude::*;
+
+use crate::{
+    config::Config,
+    service::finalized_state::{FinalizedBlock, FinalizedState},
+};
+
+use self::assert_eq;
+
+#[test]
+fn blocks_with_v5_transactions() -> Result<()> {
+    zebra_test::init();
+
+    let genesis_block: Arc<Block> =
+        zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES.zcash_deserialize_into()?;
+    let test_block: Arc<Block> = zebra_test::vectors::GENERATED_V5_1.zcash_deserialize_into()?;
+
+    let mut state = FinalizedState::new(&Config::ephemeral(), Network::Mainnet);
+
+    let hash = state
+        .commit_finalized_direct(FinalizedBlock::from(genesis_block.clone()))
+        .unwrap();
+    assert_eq!(Some(Height(0)), state.finalized_tip_height());
+    assert_eq!(hash, genesis_block.hash());
+
+    let hash = state
+        .commit_finalized_direct(FinalizedBlock::from(test_block.clone()))
+        .unwrap();
+    assert_eq!(Some(Height(1)), state.finalized_tip_height());
+    assert_eq!(hash, test_block.hash());
```

Comment:
Inside the proptest block:
1. Create the new state
2. Commit each block in the `Vec<Block>` in order, using a loop
3. Replace `assert_eq!` with `prop_assert_eq!`

You'll need to rebase or cherry-pick PR #2208 to make the block commits work - otherwise the state will panic.

Reply:
I made some changes at https://github.com/ZcashFoundation/zebra/pull/2185/commits/40b1e15694842ed32ad94ddd1ca7a75f2f67942f but not exactly as requested.

Maybe not perfect but i think is some progress as the test will generate a random block each time it runs (but only 1).

If we want a vector of blocks in order for this we need to modify the `previous_block_hash` of the `header` of each generated block to be accurate with the last one in the state which is a bit complicated (at least i was not able to make it today) .

Reply:
> If we want a vector of blocks in order for this we need to modify the `previous_block_hash` of the `header` of each generated block to be accurate with the last one in the state which is a bit complicated (at least i was not able to make it today) .

You were right, it was very complicated. I tried a few different ways before I got it working.